diff --git a/dist/core/routing/util.js b/dist/core/routing/util.js
--- a/dist/core/routing/util.js
+++ b/dist/core/routing/util.js
@@ -118,16 +118,17 @@ export function convertRes(res) {
  * @__PURE__
  */
 export function convertToQueryString(query) {
-    const queryStrings = [];
+    const searchParams = new URLSearchParams();
     Object.entries(query).forEach(([key, value]) => {
         if (Array.isArray(value)) {
-            value.forEach((entry) => queryStrings.push(`${key}=${entry}`));
+            value.forEach((entry) => searchParams.append(key, entry));
         }
         else {
-            queryStrings.push(`${key}=${value}`);
+            searchParams.append(key, value);
         }
     });
-    return queryStrings.length > 0 ? `?${queryStrings.join("&")}` : "";
+    const queryString = searchParams.toString();
+    return queryString ? `?${queryString}` : "";
 }
 /**
  * Given a raw query string, returns a record with key value-array pairs
